src/Metal/MTLIntersectionFunctionTable.cs

```diff
@@ -12,13 +12,23 @@ public static class MTLIntersectionFunctionTableExtensions {
 		[Mac (11,0), iOS (14,0), NoTV]
 		public static void SetBuffers (this IMTLIntersectionFunctionTable table, IMTLBuffer[] buffers, nuint[] offsets, NSRange range)
 		{
-			// pin address, pass it as a IntPtr which is what the protocol expects
-			var offsetsHandle = GCHandle.Alloc (offsets, GCHandleType.Pinned);
-			IntPtr offsetsPtr = offsetsHandle.AddrOfPinnedObject ();
-			try {
-				table.SetBuffers (buffers, offsetsPtr, range);
-			} finally {
-				offsetsHandle.Free ();
+			if (buffers == null)
+				throw new ArgumentNullException (nameof (buffers));
+			if (offsets == null)
+				throw new ArgumentNullException (nameof (offsets));
+
+			unsafe {
+				// cannot used fixed or we get:
+				// Cannot take the address of, get the size of, or declare a pointer to a managed type ('IMTLBuffer')
```

Comment:
You need to convert the `IMTLBuffer[]` array to an `IntPtr[]` array, and a pointer to that.